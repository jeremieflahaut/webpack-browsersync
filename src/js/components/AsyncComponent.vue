<template>
  <div>
    <button @click="fetchResults">
      Users
    </button>
    <ul>
      <li v-for="user in users">
        {{ user.id }}
      </li>
    </ul>
  </div>
</template>

<script>

export default {
  data () {
    return {
      users: null
    }
  },

  methods: {
    fetchResults () {
      this.$http.get('https://jsonplaceholder.typicode.com/users').then(response => {
        this.users = response.body
      })
    }
  }
}
</script>
