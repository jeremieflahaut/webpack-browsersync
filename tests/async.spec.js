import { shallowMount } from '@vue/test-utils'
import AsyncComponent from '../src/js/components/AsyncComponent.vue'


describe('Async', () => {
    it('users is null on created', () => {
        const wrapper = shallowMount(AsyncComponent)
        expect(wrapper.vm.users).toBe(null)
    })
    
    it('fetches async when a button is clicked', (done) => {
        const wrapper = shallowMount(AsyncComponent, {
            mocks : {
                $http : {
                    get: () => new Promise(resolve => resolve({ 
                        body: [{
                            "id": 1,
                            "name": "Leanne Graham",
                            "username": "Bret",
                            "email": "Sincere@april.biz",
                            "phone": "1-770-736-8031 x56442",
                            "website": "hildegard.org"
                        }] 
                    }))
                }
            }
        })
    
        wrapper.find('button').trigger('click')
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.users).toHaveLength(1)
            done()
          })
      })
})
