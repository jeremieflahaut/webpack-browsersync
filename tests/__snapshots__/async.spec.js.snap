// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async fetches async when a button is clicked 1`] = `
"<div><button>
    Users
  </button> <ul><li>
      Leanne Graham
    </li></ul></div>"
`;
